File: SDKs/amazonaws.com/glue/2017-03-31/go/pkg/models/operations/createconnection.go
package operations

import (
	"openapi/pkg/models/shared"
)

type CreateConnectionXAmzTargetEnum string

const (
	CreateConnectionXAmzTargetEnumAwsGlueCreateConnection CreateConnectionXAmzTargetEnum = "AWSGlue.CreateConnection"
)

type CreateConnectionHeaders struct {
	XAmzAlgorithm     *string                        `header:"name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                        `header:"name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                        `header:"name=X-Amz-Credential"`
	XAmzDate          *string                        `header:"name=X-Amz-Date"`
	XAmzSecurityToken *string                        `header:"name=X-Amz-Security-Token"`
	XAmzSignature     *string                        `header:"name=X-Amz-Signature"`
	XAmzSignedHeaders *string                        `header:"name=X-Amz-SignedHeaders"`
	XAmzTarget        CreateConnectionXAmzTargetEnum `header:"name=X-Amz-Target"`
}

type CreateConnectionRequest struct {
	Headers CreateConnectionHeaders
	Request shared.CreateConnectionRequest `request:"mediaType=application/json"`
}

type CreateConnectionResponse struct {
	AlreadyExistsException               *interface{}
	ContentType                          string
	CreateConnectionResponse             map[string]interface{}
	GlueEncryptionException              *interface{}
	InvalidInputException                *interface{}
	OperationTimeoutException            *interface{}
	ResourceNumberLimitExceededException *interface{}
	StatusCode                           int64
}

File: SDKs/amazonaws.com/glue/2017-03-31/typescript/src/sdk/models/operations/createconnection.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum CreateConnectionXAmzTargetEnum {
    AwsGlueCreateConnection = "AWSGlue.CreateConnection"
}


export class CreateConnectionHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: CreateConnectionXAmzTargetEnum;
}


export class CreateConnectionRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: CreateConnectionHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.CreateConnectionRequest;
}


export class CreateConnectionResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  alreadyExistsException?: any;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  createConnectionResponse?: Map<string, any>;

  @SpeakeasyMetadata()
  glueEncryptionException?: any;

  @SpeakeasyMetadata()
  invalidInputException?: any;

  @SpeakeasyMetadata()
  operationTimeoutException?: any;

  @SpeakeasyMetadata()
  resourceNumberLimitExceededException?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
